src/components/oscillator/test/oscillator_tests-implementation.adb

```diff
@@ -64,6 +66,18 @@ package body Oscillator_Tests.Implementation is
 
       -- Make sure parameters changed:
       Check_Parameters (2.9, 7.8, 0.4);
+
+      -- Validate a parameter:
+      Test_Param_Update.Operation := Parameter_Enums.Parameter_Operation_Type.Validate;
+      -- Set to a valid value and assert:
+      Parameter_Update_Status_Assert.Eq (T.Stage_Parameter (T.Parameters.Frequency ((Value => 5.1))), Success);
+      T.Component_Instance.Parameter_Update_T_Modify (Test_Param_Update);
+      Parameter_Update_Status_Assert.Eq (Test_Param_Update.Status, Validation_Success);
+      -- Set to an invalid value (arbitrarily forced 999.0 invalid in implementation)
+      -- assert:
+      Parameter_Update_Status_Assert.Eq (T.Stage_Parameter (T.Parameters.Frequency ((Value => 999.0))), Success);
+      T.Component_Instance.Parameter_Update_T_Modify (Test_Param_Update);
+      Parameter_Update_Status_Assert.Eq (Test_Param_Update.Status, Validation_Error);
```

Comment:
In this case, read parameter back, and make sure if 999.0.

Even better, run tick, which runs update_parameters(), and ensure that the component's parameter is 5.1 still.

Reply:
Added in [3eafa26](https://github.com/lasp/adamant_example/pull/31/commits/3eafa26fb710bde3d62750f3e401090cff24b15c), but noted the parameter is updating.

Reply:
[3ca1e26](https://github.com/lasp/adamant_example/pull/31/commits/3ca1e265193aa98703d44772aa39a1f0f2e05184) adds working parameter checks between each stage.